app/frontend/actions.ts

```diff
@@ -898,13 +926,13 @@ export default ({setState, getState}: {setState: SetStateFn; getState: GetStateF
     const state = getState()
     setPoolInfo(state)
     const poolHash = state.shelleyDelegation.selectedPool.poolHash
-    const stakingKeyRegistered = state.shelleyAccountInfo.hasStakingKey
+    const stakingKeyRegistered = sourceAccountState(state).shelleyAccountInfo.hasStakingKey
```

Comment:
naming a function with a noun i.e. `sourceAccountState` is quite misleading, something like `getSourceAccountState` would be more intuitive